src/strspan.rs

```diff
@@ -31,6 +55,11 @@ impl<'a> StrSpan<'a> {
         StrSpan { text: &text[start..end], start }
     }
 
+    /// Returns `true` is self is empty.
+    pub fn empty(&self) -> bool {
+        self.start == self.end()
```

Comment:
It should be called `is_empty`.

And `self.text.is_empty()` is more than enough.

Reply:
I would argue that `str_span.is_empty()` is still better enough than `str_span.as_str().is_empty()` to be worth it

Reply:
I was talking about `self.start == self.end()` specifically.

Reply:
I'm not sure I understand what you mean.

Reply:
```rust
pub fn is_empty(&self) -> bool {
    self.text.is_empty()
}
```